src/us/kbase/auth2/lib/Authentication.java

```diff
@@ -352,6 +358,9 @@ public Password createLocalUser(
 			// user storage call needs to throw an exception so can't use an Answer
 			clear(passwordHash);
 			clear(salt);
+			if (pwd_copy != null) {
+				Password.clearPasswordArray(pwd_copy);
```

Comment:
Maybe clearPasswordArray(null) should be a no-op? Then you don't have to check for nulls.